### PR TITLE
Tools extension guide

### PR DESCRIPTION
**Pull request by ntrogh**
Implements https://github.com/microsoft/vscode-docs/issues/7725

### CODE REVIEW

File: api/extension-guides/tools.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+- **Let the LLM take some action dynamically**. The LLM itself can't perform calculations or make calls to other systems. For example, use a tool to run a terminal command and return the output to the LLM.
+- **Hook up some context/behavior that is contributed by another VS Code extension**. For example, you might have a tool that uses the Git extension to retrieve information about the current repository.
+
+## Tool-calling flow
+
+1. Retrieve the list of relevant tools
+1. Send the request to the LLM, providing the list of tool definitions to consider
+1. The LLM generates a response, which may include one or more requests to invoke a tool
+1. Invoke the tool by using the parameter values provided in the LLM response
+1. Provide the tool responses to the LLM
```
Comment (roblourens) [inline api/extension-guides/tools.md:38]: Suggestion- "send another request to the LLM, including tool results"

File: api/extension-guides/tools.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+1. The LLM generates the final user response, which may incorporate tool responses
```
Comment (roblourens) [inline api/extension-guides/tools.md:39]: Maybe mention that this process can loop- the LLM may want to call more tools

File: api/extension-guides/tools.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+1. Return the result from the LLM. This might contain error details or tool-calling metadata.
+
+    ```ts
+    return await libResult.result;
+    ```
+
+The full source code of this [tool-calling sample](https://github.com/microsoft/vscode-extension-samples/blob/main/chat-sample/src/chatUtilsSample.ts) is available in the VS Code Extension Samples repository.
+
+### Implement tool calling with prompt-tsx
```
Comment (roblourens) [inline api/extension-guides/tools.md:91]: I'd say the other option is to implement tool calling yourself. You could use prompt-tsx for this, but you don't have to

**Review by roblourens — COMMENTED** [review]
This is great! Thanks!

**Comment by ntrogh** [comment]
@roblourens added your suggestions

**Review by roblourens — APPROVED** [review]

**Pull request merged**